packages/custom_lint/lib/basic_runner.dart

```diff
@@ -0,0 +1,179 @@
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:analyzer_plugin/protocol/protocol_generated.dart';
+import 'package:path/path.dart' as p;
+
+import 'src/analyzer_plugin/analyzer_plugin.dart';
+import 'src/analyzer_plugin/plugin_delegate.dart';
+import 'src/protocol/internal_protocol.dart';
+import 'src/runner.dart';
+
+// ignore: do_not_use_environment
+const _isReleaseMode = bool.fromEnvironment('dart.vm.product');
```

Comment:
Can we even be in release mode? Afaik I don't think code is ever run with this flag on

Reply:
What about the custom_lint executable on CI? e.g. `dart run custom_lint`

Reply:
I just checked:

Doing:

```dart
dart pub global activate -s path <path to bin>
custom_lint
```

leads to release-mode = false

On the other hand, asserts are disabled.

Reply:
Maybe we need to check for asserts for this case then instead. When run as an analyzer plugin (not via the tool/..debug.dart) does it run with asserts enabled or not?